content/en/blog/_posts/2024-04-05-diy-create-your-own-cloud-with-kubernetes-part-2/index.md

```diff
@@ -0,0 +1,150 @@
+But nowadays, this functionality might be provided as part of the CNI plugin. The most advanced implementation is offered by the [Cilium](https://cilium.io/) project, which can be run in kube-proxy replacement mode.
+
+Cilium is based on the eBPF technology, which allows for efficient offloading of the Linux networking stack, thereby improving performance and security compared to traditional methods based on iptables.
+
+In practice, Cilium and Kube-OVN can be easily [integrated](https://kube-ovn.readthedocs.io/zh-cn/stable/en/advance/with-cilium/) to provide a unified solution that offers seamless multi-tenant networking for virtual machines, as well as advanced network policies and combined services network functionality.
+
+### External Traffic Load Balancer
+
+At this stage, you already have everything needed to run virtual machines in Kubernetes. But there is actually one more thing.
+You still need to access your services from outside your cluster, and an external load balancer will help you with organizing this.
+
+For bare metal Kubernetes clusters, there are several load balancers available: [MetalLB](https://metallb.universe.tf/), [kube-vip](https://kube-vip.io/), [LoxiLB](https://www.loxilb.io/), also [Cilium](https://docs.cilium.io/en/latest/network/lb-ipam/) and [Kube-OVN](https://kube-ovn.readthedocs.io/zh-cn/latest/en/guide/loadbalancer-service/) provides built-in implementation.
+
+The role of a external load balancer is to provide a stable address available externally and direct external traffic to the services network.
+The services network plugin will direct it to your pods and virtual machines as usual.
+
+{{< figure src="net-services.svg" caption="A diagram showing the role of the external load balancer on the Kubernetes network scheme" alt="The role of the external load balancer on the Kubernetes network scheme" >}}
+
+In most cases, setting up a load balancer on bare metal is achieved by creating floating IP address on the nodes within the cluster, and announce it externally using ARP/NDP or BGP protocols.
+
+After exploring various options, we decided that MetalLB is the simplest and most reliable solution, although we do not strictly enforce the use of only it.
+
+Another benefit is that in L2 mode, MetalLB speakers continuously checking thir neighbors state by sending preforming liveness checks using a memberlist protocol.
+This enables failover that works independently of Kubernetes control-plane.
+
+## Conclusion
+
+This concludes our overview of virtualization, storage, and networking in Kubernetes. The technologies mentioned here are available and already pre-configured on the [Cozystack](https://github.com/aenix-io/cozystack) platform, where you can try them with no limitations.
+
+In the [next article](/blog/2024/04/05/diy-create-your-own-cloud-with-kubernetes-part-3/), we will detail how, on top of this, we implement the provisioning of fully functional Kubernetes clusters with just the push of a button.
```

Comment:
Does 'we' in this context, refer to a group of individuals?

We avoid using 'we' for readers. Rather, you should use 'you' for them.

Reply:
Try:
```diff
-In the [next article](/blog/2024/04/05/diy-create-your-own-cloud-with-kubernetes-part-3/), we will detail how, on top of this, we implement the provisioning of fully functional Kubernetes clusters with just the push of a button.
+In the [next article](/blog/2024/04/05/diy-create-your-own-cloud-with-kubernetes-part-3/), I'll
+detail how, on top of this, you can implement the provisioning of fully functional Kubernetes clusters
+with just the push of a button.
```

(the “we” becomes an “I” and a ”you”, which helps readers especially if English is not their first language)

Reply:
Thanks, accepted